pilot/pkg/model/authentication.go

```diff
@@ -70,7 +73,37 @@ func GetConsolidateAuthenticationPolicy(store IstioConfigStore, service *Service
 	return nil
 }
 
-// ConstructSdsSecretConfig constructs SDS Sececret Configuration.
+// ConstructSdsSecretConfig constructs SDS secret configuration for ingress gateway.
+func ConstructSdsSecretConfigForGatewayListener(name, sdsUdsPath string) *auth.SdsSecretConfig {
+	if name == "" || sdsUdsPath == "" {
+		return nil
+	}
+
+	gRPCConfig := &core.GrpcService_GoogleGrpc{
+		TargetUri:  sdsUdsPath,
+		StatPrefix: SDSStatPrefix,
+	}
+
+	return &auth.SdsSecretConfig{
+		Name: name,
+		SdsConfig: &core.ConfigSource{
+			ConfigSourceSpecifier: &core.ConfigSource_ApiConfigSource{
+				ApiConfigSource: &core.ApiConfigSource{
+					ApiType: core.ApiConfigSource_GRPC,
+					GrpcServices: []*core.GrpcService{
+						{
+							TargetSpecifier: &core.GrpcService_GoogleGrpc_{
```

Comment:
Why google Grpc ? We use regular envoy grpc for ADS. 

I don't think we should mix, is there a specific reason ? If yes, we should switch ADS too.

Reply:
@quanjielin Hi Quanjie, do you know more about this?

Reply:
```The Envoy gRPC client is a minimal custom implementation of gRPC that makes use of Envoy’s HTTP/2 upstream connection management. Services are specified as regular Envoy clusters``` according to https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/grpc#grpc-services, while for envoy->nodeagent we use google grpc since need to pass token as call_credential /cc @qiwzhang